www/perl-lib/SMT/Rest/SCCAPIv4.pm

```diff
@@ -359,6 +359,23 @@ sub delete_single_product
     my $guid = $self->user();
     my $c    = JSON::decode_json($self->read_post());
 
+    #
+    # lookup the Clients target
+    #
+    my $target = SMT::Utils::lookupTargetForClient($self->dbh(), $guid, $self->request());
+
+    #
+    # find Catalogs
+    #
+    my $existingregs = SMT::Utils::lookupRegistrationByGUID($self->dbh(), $guid, $self->request());
+    my @pidarr = keys %{$existingregs};
+    my $catalogs = SMT::Registration::findCatalogs($self->request(), $self->dbh(), $target, \@pidarr);
```

Comment:
It looks like you moved the check up from the bottom of the method.  However, if I understand correctly, in its old position, it was looking at the number of catalogs(repositories) remaining after the deletion of the requested registration.  In this position, will it be looking at the number of catalogs present before deletion?  If this is true, shouldn't it then calculate how many catalogs would be present if the requested registration was deleted, and error if that is zero?